File: src/client/java/com/williambl/haema/client/vampire_mobs/VampiricZombieRenderer.java
package com.williambl.haema.client.vampire_mobs;

import net.minecraft.client.model.geom.ModelLayerLocation;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.client.renderer.entity.ZombieRenderer;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.monster.Zombie;

import static com.williambl.haema.Haema.id;

public class VampiricZombieRenderer extends ZombieRenderer {
    private static final ResourceLocation TEXTURE_LOCATION = id("textures/entity/vampiric_zombie.png");

    public VampiricZombieRenderer(EntityRendererProvider.Context context) {
        super(context);
    }

    @Override
    public ResourceLocation getTextureLocation(Zombie zombie) {
        return TEXTURE_LOCATION;
    }
}

File: src/main/kotlin/com/williambl/haema/client/VampiricZombieRenderer.kt
package com.williambl.haema.client

import com.williambl.haema.id
import net.minecraft.client.render.entity.EntityRendererFactory
import net.minecraft.client.render.entity.ZombieEntityRenderer
import net.minecraft.entity.mob.ZombieEntity
import net.minecraft.util.Identifier

class VampiricZombieRenderer(context: EntityRendererFactory.Context) : ZombieEntityRenderer(context) {
    override fun getTexture(zombieEntity: ZombieEntity?): Identifier = texture

    companion object {
        val texture = id("textures/entity/vampiric_zombie.png")
    }
}